src/extensions/agent_modules/cloud_communication/communication_channel/mqtt_broker/src/adu_communication_channel.c

```diff
@@ -577,24 +351,33 @@ void ADUC_CommunicationChannel_OnMessage(
     Log_Info("Topic: %s; QOS: %d; mid: %d", msg->topic, msg->qos, msg->mid);
 
     // All parameters are required.
-    if (mosq == NULL || obj == NULL || msg == NULL || props == NULL)
+    if (mosq == NULL || obj == NULL || msg == NULL)
     {
         Log_Error("Null parameter (mosq=%p, obj=%p, msg=%p, props=%p)", mosq, obj, msg, props);
         goto done;
     }
 
-    // Get the message type. (mt = message type)
-    if (!ADU_mosquitto_read_user_property_string(props, "mt", &msg_type) || IsNullOrEmpty(msg_type))
+    if (s_commMgrState.mqttSettings.mqttVersion == MQTT_PROTOCOL_V5)
```

Comment:
not a fan of the all the static-to-file vars. Would've been nice to have a thin C interface that dips into c++ impl and more modular with obj instances.